keps/sig-node/4265-proc-mount/README.md

```diff
@@ -0,0 +1,524 @@
+
+Consider the following in developing an upgrade/downgrade strategy for this
+enhancement:
+- What changes (in invocations, configurations, API use, etc.) is an existing
+  cluster required to make on upgrade, in order to maintain previous behavior?
+- What changes (in invocations, configurations, API use, etc.) is an existing
+  cluster required to make on upgrade, in order to make use of the enhancement?
+-->
+Turn off the feature gate to turn off the feature.
+
+### Version Skew Strategy
+
+<!--
+If applicable, how will the component handle version skew with other
+components? What are the guarantees? Make sure this is in the test plan.
+
+Consider the following in developing a version skew strategy for this
+enhancement:
+- Does this enhancement involve coordinating behavior in the control plane and nodes?
+- How does an n-3 kubelet or kube-proxy without this feature available behave when this feature is used?
+- How does an n-1 kube-controller-manager or kube-scheduler without this feature available behave when this feature is used?
+- Will any other components on the node change? For example, changes to CSI,
+  CRI or CNI may require updating that component before the kubelet.
+-->
+
+The feature gate is only processed by the API server--Kubelet has no awareness of it. API server will scrub the ProcMount field from the request
+if it doesn't support the feature gate. Since all supported Kubelet versions support ProcMountType field, there's no version skew worry.
+API server can have the feature gate toggled without worrying about doing the same for Kubelets.
+
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+- [x] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name: ProcMountType
+  - Components depending on the feature gate: kube-apiserver (kube-apiserver filters `procMount` field if it's not enabled).
+
+###### Does enabling the feature change any default behavior?
+
+No, only gives a user access to the Unmasked ProcMountType
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+Yes. This can be done by removing the feature gate from all kube-apiservers. To fully roll back, the nodes will need to be drained or rebooted,
+as the Kubelet will not remove the `procMount` of an already running container.
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+Nothing special. The pod's `procMount` field depends on where in the enablement process the kube-apiserver was when it was created.
+The container has to be restarted to be up to date with the kube-apiserver.
+
+###### Are there any tests for feature enablement/disablement?
+
+Yes. I have manually tested feature enablement and disablement on kube-apiserver, and verified that pods are not recreated without
+a drain. There will be an e2e test to verify this as well.
```

Comment:
if this ends up impractical in practice a unit test showing the field stripping will be sufficient along with a manual test of disablement.